pct.sh

```diff
@@ -55,5 +55,7 @@ rm -fv pct-work/durable-task/target/surefire-reports/TEST-org.jenkinsci.plugins.
 rm -fv pct-work/git-client/target/surefire-reports/TEST-hudson.plugins.git.GitExceptionTest.xml
 # TODO fails for one reason in (non-PCT) official sources, run locally; and for another reason in PCT in Docker; passes in official sources in Docker, or locally in PCT
 rm -fv pct-work/git-client/target/surefire-reports/TEST-org.jenkinsci.plugins.gitclient.FilePermissionsTest.xml
+# TODO fails with NPE at at org.jenkinsci.plugins.workflow.libs.LibraryMemoryTest.loaderReleased(LibraryMemoryTest.java:85)
```

Comment:
Will need to track down why.

Reply:
Agreed that this needs to be tracked down, but could you please clarify whether you think this needs to be tracked down in general or tracked down _as a prerequisite_ to merging this PR?

Once this project is stable, I think it makes sense to have a policy where we don't accept plugin bumps in the BOM until that plugin's tests are green or a PR has been opened getting them to be green. The reason why I think this makes sense is that bumping a plugin whose tests _were_ green to a version where _some_ tests are failing would be a regression. In the worst case this could provide consumers of the BOM with buggy plugins, which is certainly undesirable.

On the other hand, I'm not so sure a policy like this necessarily makes sense during the initial bringup phase of the project. We don't yet have a baseline for all plugin versions against which we can measure regressions. In practice, Jenkins administrators like myself [upgrade and brace for impact](https://twitter.com/directhex/status/1155896955526176770) on a weekly basis. As we bring up this project, I think it would be useful to add as many plugins as possible to the BOM, even if there are some existing PCT test failures that still need to be tracked down. Doing so would unblock rapid development of the BOM and make it more useful to potential consumers while also establishing a baseline against which we can measure future regressions. It would also allow us to get a sense of the existing issues with which we have to contend _earlier on_, which could help us prioritize them and form a TODO list that could be parallelized among several developers (including volunteers like myself!). As the project matures, our focus could then shift from rapid expansion to stabilization by fixing all the existing issues we discovered during the first phase, as well as being more strict about saying no to any bumps that would regress our (by then, known) baseline.

There is some precedent for a policy like this in your existing commits to the project. For example, in b7d411e0 you added a TODO regarding [a test failure](https://github.com/jenkinsci/bom/blob/b7d411e0/pct.sh#L56-L57) for which a cause doesn't yet seem to be known and a fix doesn't yet seem to be available. This seems to match my views above in that it expands plugin coverage during the bringup phase of the project even though the test results aren't yet green.

What do you think?

Reply:
Generally that seems fine, so long as there is some expectation that all exclusions do get triaged. The one case without a PR is one I am still unsure of—feels like an environmental issue and the test could just be deleted. (I also heard of the same test failing in an unrelated PCT run.)

Reply:
I _do_ prefer to establish with some level of confidence that failures are technical test issues and not in fact incompatibilities between selected plugin versions.

Reply:
> Will need to track down why.

Done in jenkinsci/workflow-durable-task-step-plugin#117.